src/pybindings/pycontroller/ChipDeviceCtrl.py

```diff
@@ -0,0 +1,320 @@
+        )
+
+    def ResolveNode(self, fabricid, nodeid):
+        return self._ChipStack.CallAsync(
+            lambda: self._dmLib.pychip_Resolver_ResolveNode(fabricid, nodeid)
+        )
+
+    def GetAddressAndPort(self, nodeid):
+        address = create_string_buffer(64)
+        port = c_uint16(0)
+
+        error = self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_GetAddressAndPort(
+                self.devCtrl, nodeid, address, 64, pointer(port))
+        )
+
+        return (address.value.decode(), port.value) if error == 0 else None
+    def DiscoverCommissioningLongDiscriminator(self, long_discriminator):
+        return self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_DiscoverCommissioningLongDiscriminator(self.devCtrl, long_discriminator)
+        )
+
+    def PrintDiscoveredDevices(self):
+        return self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_PrintDiscoveredDevices(self.devCtrl)
+        )
+    def PrintSetupPayload(setup_payload):
+        print("Version: {}".format(setup_payload.version))
+        print("Vendor ID: {}".format(setup_payload.vendorID))
+        print("Product ID: {}".format(setup_payload.productID))
+        print("Commissioning Flow: {}".format(setup_payload.commissioningFlow))
+        print("Rendezvous Information Flag: {}".format(setup_payload.rendezvousInformation.Raw()))
+        print("Discriminator: {}".format(setup_payload.discriminator))
+        print("Setup Pin Code: {}".format(setup_payload.setUpPINCode))
+        print("All Vendor Options: {}".format(setup_payload.getAllOptionalVendorData()))
+
+    def ParseQRCode(self, qrCode):
+        setup_payload = SetupPayload()
+        qr_code_parser = QRCodeSetupPayloadParser(qrCode)
+        qr_code_parser.populatePayload(setup_payload)
+
+        return setup_payload
+
+    def ParseManualSetupCode(self, code):
+        setup_payload = SetupPayload()
+        qr_code_parser = ManualSetupPayloadParser(code)
+        qr_code_parser.populatePayload(setup_payload)
+        return setup_payload
+
+    def GetIPForDiscoveredDevice(self, idx, addrStr, length):
+        return self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_GetIPForDiscoveredDevice(self.devCtrl, idx, addrStr, length)
+        )
+
+    def DiscoverAllCommissioning(self):
+        return self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_DiscoverAllCommissioning(self.devCtrl)
+        )
+
+    def GetFabricId(self):
+        fabricid = c_uint64(0)
+
+        error = self._ChipStack.Call(
+            lambda: self._dmLib.pychip_DeviceController_GetFabricId(
+                self.devCtrl, pointer(fabricid))
+        )
+
+        if error == 0:
+            return fabricid.value
+        else:
+            return 0
+
+    def ZCLSend(self, cluster, command, nodeid, endpoint, groupid, args, blocking=False):
```

Comment:
please consider to replace ZCL with IM in function naming

Reply:
I can update, but the python files are just place holders in order to make this in parity with the previous implementation. Here is the doc for what needs to be completed in the python files: https://docs.google.com/document/d/1BKTfi2icKrWl5-VEfpXceqrqtkGg7luVBPmBNbDMExk